be/src/vec/aggregate_functions/aggregate_function_java_udaf.h

```diff
@@ -0,0 +1,358 @@
+        env->CallNonvirtualBooleanMethod(executor_obj, executor_cl, executor_result_id,           \
+                                         to.size() - 1);                                          \
+    } else {                                                                                      \
+        return Status::InvalidArgument(strings::Substitute(                                       \
+                "Java UDAF doesn't support return type is $0 now !", result_type->get_name()));   \
+    }
+#endif
+            EVALUATE_JAVA_UDAF;
+        } else {
+            *output_null_value = -1;
+            *output_value_buffer = reinterpret_cast<int64_t>(to.get_raw_data().data);
+            auto& data_col = to;
+            EVALUATE_JAVA_UDAF;
+            env->CallNonvirtualBooleanMethod(executor_obj, executor_cl, executor_result_id,
+                                             to.size() - 1);
+        }
+        return JniUtil::GetJniExceptionMsg(env);
+    }
+
+private:
+    Status register_func_id(JNIEnv* env) {
+        auto register_id = [&](const char* func_name, const char* func_sign, jmethodID& func_id) {
+            func_id = env->GetMethodID(executor_cl, func_name, func_sign);
+            Status s = JniUtil::GetJniExceptionMsg(env);
+            if (!s.ok()) {
+                return Status::InternalError(
+                        strings::Substitute("Java-Udaf register_func_id meet error and error is $0",
+                                            s.get_error_msg()));
+            }
+            return s;
+        };
+
+        RETURN_IF_ERROR(register_id("<init>", UDAF_EXECUTOR_CTOR_SIGNATURE, executor_ctor_id));
+        RETURN_IF_ERROR(register_id("add", UDAF_EXECUTOR_ADD_SIGNATURE, executor_add_id));
+        RETURN_IF_ERROR(register_id("close", UDAF_EXECUTOR_CLOSE_SIGNATURE, executor_close_id));
+        RETURN_IF_ERROR(register_id("merge", UDAF_EXECUTOR_MERGE_SIGNATURE, executor_merge_id));
+        RETURN_IF_ERROR(
+                register_id("serialize", UDAF_EXECUTOR_SERIALIZE_SIGNATURE, executor_serialize_id));
+        RETURN_IF_ERROR(
+                register_id("getValue", UDAF_EXECUTOR_RESULT_SIGNATURE, executor_result_id));
+        return Status::OK();
+    }
+
+private:
+    jclass executor_cl;
+    jobject executor_obj;
+    jmethodID executor_ctor_id;
+
+    jmethodID executor_add_id;
+    jmethodID executor_merge_id;
+    jmethodID executor_serialize_id;
+    jmethodID executor_result_id;
+    jmethodID executor_close_id;
+
+    std::unique_ptr<int64_t[]> input_values_buffer_ptr;
+    std::unique_ptr<int64_t[]> input_nulls_buffer_ptr;
+    std::unique_ptr<int64_t[]> input_offsets_ptrs;
+    std::unique_ptr<int64_t> output_value_buffer;
+    std::unique_ptr<int64_t> output_null_value;
+    std::unique_ptr<int64_t> output_offsets_ptr;
+    std::unique_ptr<int64_t> output_intermediate_state_ptr;
+
+    bool first_init;
```

Comment:
I think this variable name is a bit curious. `init` also means `first`. Maybe rename to `inited` or `first_called` ?